### DISCUSSION TITLE
USB Floppy drive support ?

### DISCUSSION THREAD

**Discussion by dherrendoerfer** [rsta2/circle#355] [Ideas]
I was trying to connect a USB Floppy disk drive using USB Mass Storage and it didn't work.
The device is detected, and if I remember correctly - it should (with a disk inserted) work with normal USB Mass Storage drivers.
What is it that needs to be done to make that work somehow ?
Since I'm trying to build an 8-bit computer, floppy support would be excellent !
Cheers !
Dirk

**Comment by rsta2** [comment]
I think this affects the following specifications, which have to be implemented:

* https://www.usb.org/sites/default/files/Mass_Storage_Specification_Overview_v1.4_2-19-2010.pdf
* https://usb.org/sites/default/files/usbmass-ufi10.pdf
* https://usb.org/sites/default/files/usb_msc_cbi_1.1.pdf

Currently Circle provides a driver for USB mass-storage devices with "Bulk-Only Transport" only. I suppose, this is not sufficient for floppy disk access.

I cannot make promises, but I will have a closer look at this, when I have some time left.

Rene
